### DISCUSSION TITLE
Giving type 6 error while decrypting

### DISCUSSION THREAD

**Discussion by amank575** [openpgpjs/openpgpjs#1579] [Q&A]
Hi , I want to read my private armored key which I have locally and decrypt .xlsx (excel table row data) which is encrypted from public key. I tried converting armored message to binary and decrypt but throws error .Any help or thread related to it is much appreciated.
```diff
const openpgp = require('openpgp');
const fs = require('fs');


async function decrypt() {
    const passphrase = 'xxxxx';
    const encryptedData = fs.createReadStream('./encrypted.xlsx.pgp','binary');
    const privateKeyBlock = fs.readFileSync('./private_key.asc','utf8');
    const privateKey = await openpgp.readPrivateKey({ armoredKey: privateKeyBlock });
    await privateKey.decrypt(passphrase);

    const decrypted = await openpgp.decrypt({
        message: await openpgp.readMessage({ binaryMessage: encryptedData }),
        privateKeys: [privateKey],
        format: 'binary',
        config: { allowInsecureDecryptionWithSigningKeys: true, allowUnauthenticatedMessages: true }
    });

    const readStream = decrypted.data;
    console.log(readStream);

}
decrypt();
```
Error
```diff
Error: Unexpected packet type: 6
    at uc.packetListToStructure (C:\Users\AmanKumar\Desktop\seerportalQAJS\node_modules\←[4mopenpgp←[24m\dist\node\openpgp.min.js:2:320617)
    at new uc (C:\Users\AmanKumar\Desktop\seerportalQAJS\node_modules\←[4mopenpgp←[24m\dist\node\openpgp.min.js:2:329923)
    at Object.exports.readPrivateKey (C:\Users\AmanKumar\Desktop\seerportalQAJS\node_modules\←[4mopenpgp←[24m\dist\node\openpgp.min.js:16:142738)
←[90m    at processTicksAndRejections (node:internal/process/task_queues:96:5)←[39m
    at async decrypt ←[90m(C:\Users\AmanKumar\Desktop\seerportalQAJS\mjs\←[39mexcelDecrypt.js:9:24←[90m)←[39m
```

**Comment by twiss** [comment]
Hi @amank575, packet type 6 is a public-key packet. So, it's not a private key, and trying to decrypt a message with it won't work. Probably the error message here could be improved, though.

**Reply by amank575** [reply]
Hi @twiss but its a private key when I tried to decrypt csv type file with the same it works fine and I am able to decrypt file with PGP tool with the same key.

**Reply by twiss** [reply]
It's possible that the file contains both a public key and a private key, perhaps? You could try to split them up, and pass only the private key.

Alternatively, one thing you could try is passing `config: { ignoreMalformedPackets: true }` to `readPrivateKey` (even though in this case it's not really malformed but rather unexpected, but it might still solve your issue).

**Reply by twiss** [reply]
Or, actually, the other option is using `readKey` (or `readKeys`), that one allows reading both public and private keys.

**Reply by amank575** [reply]
> Or, actually, the other option is using `readKey` (or `readKeys`), that one allows reading both public and private keys.
hi @twiss have created a test repo and made the said changes but still getting the same error can you look through the process and suggest were I am going wrong or how I can debug. Thanks in advance
git clone url : https://github.com/amank575/testDecrypt.git

**Reply by twiss** [reply]
The "private key" file indeed contains both a private key and a public key. Actually `readKey` indeed still doesn't work, but `readKeys` should work, and will give you both the private and public key (in an array).

**Reply by amank575** [reply]
Able to solve the problem thanks for help @twiss
